csharp/extractor/Semmle.Extraction.CSharp/Entities/Types/Type.cs

```diff
@@ -25,6 +25,25 @@ public static bool ConstructedOrParentIsConstructed(INamedTypeSymbol symbol)
                 symbol.ContainingType is not null && ConstructedOrParentIsConstructed(symbol.ContainingType);
         }
 
+        /// <summary>
+        /// Returns true in case we suspect this is a broken type.
+        /// </summary>
+        /// <param name="symbol">Type symbol</param>
+        private bool IsBrokenType(ITypeSymbol symbol)
+        {
+            if (!Context.ExtractionContext.IsStandalone ||
+                !symbol.FromSource() ||
+                symbol.IsAnonymousType)
+            {
+                return false;
+            }
+
+            // (1) public class { ... } is a broken type as it doesn't have a name.
+            // (2) public class var { ... } is an allowed type, but it overrides the `var` keyword for all uses.
+            //     It is probably a better heuristic to treat it as a broken type.
+            return string.IsNullOrEmpty(symbol.Name) || symbol.Name == "var";
```

Comment:
Are there other names that should be treated the same way? For example `dynamic`, `nint`, `nuint`?

Reply:
Yes, we can definitely extend the list with more "reserved" names.